be/src/exprs/split_part.cpp

```diff
@@ -24,6 +24,96 @@
 
 namespace starrocks {
 
+static void split_positive_index(ColumnBuilder<TYPE_VARCHAR> res, Slice& delimiter, int32_t& part_number,
```

Comment:
to be simpler, no need to add `ColumnBuilder<TYPE_VARCHAR> res` as a parameter? how about 
`static Slice split_positive_index( Slice& haystack, Slice& delimiter, int32_t part_number)` instead?